packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js

```diff
@@ -0,0 +1,226 @@
+  throw err;
+});
+
+
+MeetingInfoUtil.getDestinationType = async (from) => {
+  const {destination, type, webex} = from;
+
+  if (type) {
+    return {
+      destination,
+      type
+    };
+  }
+  const options = {};
+  const hydraId = MeetingInfoUtil.getHydraId(destination);
+
+  if (MeetingInfoUtil.isMeetingLink(destination)) {
+    LoggerProxy.logger.warn('Meeting-info:util#generateOptions --> WARN, use of Meeting Link is deprecated, please use a SIP URI instead');
+
+    options.type = _MEETING_LINK_;
+    options.destination = destination;
+  }
+  else if (MeetingInfoUtil.isSipUri(destination)) {
+    options.type = _SIP_URI_;
+    options.destination = destination;
+  }
+  else if (MeetingInfoUtil.isPhoneNumber(destination)) {
+    options.type = _SIP_URI_;
+    options.destination = destination;
+  }
+  else if (MeetingInfoUtil.isConversationUrl(destination, webex)) {
+    options.type = _CONVERSATION_URL_;
+    options.destination = destination;
+  }
+  else if (hydraId.people) {
+    options.type = _SIP_URI_;
+
+    return MeetingInfoUtil.getSipUriFromHydraPersonId(hydraId.destination, webex).then((res) => {
+      options.destination = res;
+
+      // Since hydra person ids require a unique case in which they are
+      // entirely converted to a SIP URI, we need to set a flag for detecting
+      // this type of destination.
+      options.wasHydraPerson = true;
+
+      return Promise.resolve(options);
+    });
+  }
+  else if (hydraId.room) {
+    options.type = _CONVERSATION_URL_;
+    try {
+      await webex.internal.services.waitForCatalog('postauth');
+
+      const conversationUrl = webex.internal.conversation.getUrlFromClusterId({
+        cluster: hydraId.cluster,
+        id: hydraId.destination
+      });
+
+      options.destination = conversationUrl;
+    }
+    catch (e) {
+      LoggerProxy.logger.error(`Meeting-info:util#generateOptions --> ${e}`);
+      throw (e);
+    }
+  }
+  else {
+    LoggerProxy.logger.warn('Meeting-info:util#generateOptions --> (\'MeetingInfo is fetched with meeting link, sip uri, phone number, hydra room id, hydra people id, or a conversation url.');
+    throw new ParameterError('MeetingInfo is fetched with meeting link, sip uri, phone number, hydra room id, hydra people id, or a conversation url.');
+  }
+
+  return Promise.resolve(options);
+};
+
+/**
+ * Helper function to build up a correct locus url depending on the value passed
+ * @param {Object} options type and value to fetch meeting info
+ * @param {String} options.type One of [SIP_URI, PERSONAL_ROOM, MEETING_ID, CONVERSATION_URL, LOCUS_ID, MEETING_LINK]
+ * @param {Object} options.destination ?? value.value
+ * @returns {Object} returns an object with {resource, method}
+ */
+MeetingInfoUtil.getRequestBody = (options) => {
+  const {type, destination} = options;
+  const body = {};
+
+  switch (type) {
```

Comment:
https://wiki.cisco.com/display/HFWEB/MeetingInfo+API also allows meetingUUID to be used as an input parameter and it's not covered here, should it be?

Reply:
we officially dont say we support  it but will add it for the web client , As we dont have it in the official documentation